### PR TITLE
[DEVOPS-1822] - Upload  Mobile Beta Native Build

### PR DESCRIPTION
**Pull request by Eeebru**
## Type of change
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [x] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

- [Card](https://bitwarden.atlassian.net/browse/DEVOPS-1822)

## Code changes

* **.github/secrets/*:** Adds native beta profile secrets
* **.github/workflows/build-beta.yml:** Adds beta build workflow

## Before you submit
- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: .github/workflows/build-beta.yml [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up CLOC
+        run: |
+          sudo apt-get update
+          sudo apt-get -y install cloc
+
+      - name: Print lines of code
+        run: cloc --vcs git --exclude-dir Resources,store,test,Properties --include-lang C#,XAML
```
Comment (mimartin12) [inline .github/workflows/build-beta.yml:32]: ⛏️  We no longer use cloc jobs, they have started to be removed from other workflows. https://github.com/bitwarden/clients/pull/7893
```suggestion

```

Comment (Eeebru) [reply .github/workflows/build-beta.yml:32]: Thanks

File: .github/workflows/build-beta.yml [hunk 2]
```diff
@@ -3,3 +3,356 @@ name: Build Beta
...
+
+          WATCH_APP_UUID=$(grep UUID -A1 -a $WATCH_APP_PROFILE_PATH | grep -io "[-A-F0-9]\{36\}")
+          cp $WATCH_APP_PROFILE_PATH "$PROFILES_DIR_PATH/$WATCH_APP_UUID.mobileprovision"
+
+          WATCH_APP_EXTENSION_UUID=$(grep UUID -A1 -a $WATCH_APP_EXTENSION_PROFILE_PATH | grep -io "[-A-F0-9]\{36\}")
+          cp $WATCH_APP_EXTENSION_PROFILE_PATH "$PROFILES_DIR_PATH/$WATCH_APP_EXTENSION_UUID.mobileprovision"
+
+      - name: Restore packages
+        run: |
+          dotnet restore
```
Comment (mimartin12) [inline .github/workflows/build-beta.yml:205]: ⛏️  linting
```suggestion
      - name: Restore packages
        run:  dotnet restore
```

File: .github/workflows/build-beta.yml [hunk 3]
```diff
@@ -3,3 +3,356 @@ name: Build Beta
...
+          ARCHIVE_PATH: ./${{ env.main_app_folder_path }}/bin/Debug/${{ env.target-net-version }}-ios/iossimulator-x64
+          EXPORT_PATH: ./bitwarden-export
+        run: |
+          zip -r -q ${{ env.app_ci_output_filename }}.app.zip $ARCHIVE_PATH
+          mv ${{ env.app_ci_output_filename }}.app.zip $EXPORT_PATH
+
+      - name: Show Bitwarden Export
+        shell: bash
+        run: |
+          ls -a -R ./bitwarden-export
```
Comment (mimartin12) [inline .github/workflows/build-beta.yml:261]: ⛏️  Linting
```suggestion
      - name: Show Bitwarden Export
        shell: bash
        run:  ls -a -R ./bitwarden-export
```

**Review by mimartin12 — CHANGES_REQUESTED** [review]

File: .github/workflows/build-beta.yml [hunk 4]
```diff
@@ -3,3 +3,356 @@ name: Build Beta
...
+
+          WATCH_APP_EXTENSION_UUID=$(grep UUID -A1 -a $WATCH_APP_EXTENSION_PROFILE_PATH | grep -io "[-A-F0-9]\{36\}")
+          cp $WATCH_APP_EXTENSION_PROFILE_PATH "$PROFILES_DIR_PATH/$WATCH_APP_EXTENSION_UUID.mobileprovision"
+
+      - name: Restore packages
+        run: |
+          dotnet restore
+
+      - name: Restore tools
+        run: dotnet tool restore
```
Comment (mimartin12) [inline .github/workflows/build-beta.yml]: 💭 Combining would make more sense, thoughts?
```suggestion
      - name: Restore dependencies
        run: |
          dotnet restore
          dotnet tool restore
```

**Review by mimartin12 — APPROVED** [review]

**Review by urbinaalex17 — APPROVED** [review]

**Review by vvolkgang — APPROVED** [review]

**Pull request merged**